### ISSUE TITLE
Doesn't support wxWidgets 3

### ISSUE THREAD

**Issue by niffiwan** [scp-fs2open/PCS2#9]
Or at least seems to not and wx 2.8 is no longer available in ubuntu & derivatives. Should AppImage the sucker.

Anyway; here's the compile errors:

```
/home/mememe/src/PCS2/src/Main_Panel.cpp: In constructor ‘main_panel::main_panel(wxFrame*)’:
/home/mememe/src/PCS2/src/Main_Panel.cpp:190:9: error: ‘wxGA_PROGRESSBAR’ was not declared in this scope
         wxGA_PROGRESSBAR | wxGA_SMOOTH);
         ^
/home/mememe/src/PCS2/src/Main_Panel.cpp:199:23: warning: ‘void wxSplitterWindow::SetSashSize(int)’ is deprecated [-Wdeprecated-declarations]
  split1->SetSashSize(3);
                       ^
In file included from /usr/include/wx-3.0/wx/wx.h:14:0,
                 from /home/mememe/src/PCS2/src/main_panel.h:60,
                 from /home/mememe/src/PCS2/src/Main_Panel.cpp:120:
/usr/include/wx-3.0/wx/generic/splitter.h:225:31: note: declared here
     wxDEPRECATED_INLINE( void SetSashSize(int WXUNUSED(width)), return; )
                               ^
/usr/include/wx-3.0/wx/defs.h:615:43: note: in definition of macro ‘wxDEPRECATED’
 #define wxDEPRECATED(x) wxDEPRECATED_DECL x
                                           ^
/usr/include/wx-3.0/wx/generic/splitter.h:225:5: note: in expansion of macro ‘wxDEPRECATED_INLINE’
     wxDEPRECATED_INLINE( void SetSashSize(int WXUNUSED(width)), return; )
     ^
/home/mememe/src/PCS2/src/Main_Panel.cpp:260:24: warning: ‘void wxSplitterWindow::SetSashSize(int)’ is deprecated [-Wdeprecated-declarations]
   split2->SetSashSize(3);
                        ^
In file included from /usr/include/wx-3.0/wx/wx.h:14:0,
                 from /home/mememe/src/PCS2/src/main_panel.h:60,
                 from /home/mememe/src/PCS2/src/Main_Panel.cpp:120:
/usr/include/wx-3.0/wx/generic/splitter.h:225:31: note: declared here
     wxDEPRECATED_INLINE( void SetSashSize(int WXUNUSED(width)), return; )
                               ^
/usr/include/wx-3.0/wx/defs.h:615:43: note: in definition of macro ‘wxDEPRECATED’
 #define wxDEPRECATED(x) wxDEPRECATED_DECL x
                                           ^
/usr/include/wx-3.0/wx/generic/splitter.h:225:5: note: in expansion of macro ‘wxDEPRECATED_INLINE’
     wxDEPRECATED_INLINE( void SetSashSize(int WXUNUSED(width)), return; )
     ^
/home/mememe/src/PCS2/src/Main_Panel.cpp: In member function ‘void main_panel::on_load_chunk(wxCommandEvent&)’:
/home/mememe/src/PCS2/src/Main_Panel.cpp:685:89: error: ‘wxOPEN’ was not declared in this scope
  wxFileDialog fdlg(NULL, _("Select Import File"), _(""), _(""), PCS2_SUPPORTED_FORMATS, wxOPEN | wxFILE_MUST_EXIST);
                                                                                         ^
/home/mememe/src/PCS2/src/Main_Panel.cpp:685:98: error: ‘wxFILE_MUST_EXIST’ was not declared in this scope
  wxFileDialog fdlg(NULL, _("Select Import File"), _(""), _(""), PCS2_SUPPORTED_FORMATS, wxOPEN | wxFILE_MUST_EXIST);
                                                                                                  ^
src/CMakeFiles/pcs2.dir/build.make:326: recipe for target 'src/CMakeFiles/pcs2.dir/Main_Panel.cpp.o' failed
make[2]: *** [src/CMakeFiles/pcs2.dir/Main_Panel.cpp.o] Error 1
CMakeFiles/Makefile2:85: recipe for target 'src/CMakeFiles/pcs2.dir/all' failed
make[1]: *** [src/CMakeFiles/pcs2.dir/all] Error 2
Makefile:83: recipe for target 'all' failed
make: *** [all] Error 2
```

**Comment by niffiwan** [comment]
hahaha, I've just completed some changes to support wx3... and _then_ I discover that there's two PRs already in existence that do exactly that >.>

I might see if I can use #8 as a base for the stuff I did (I wasn't thinking of maintaining wx2.8 support), which included merging in [fmt](http://fmtlib.net) to deal with the Format issues without relying on platform specific printf options, as well as being a test for integration it into FSO as well...

**Comment by niffiwan** [comment]
ok, see #10 for my effort at this, I didn't end up using #8 as a base, I decided there's not much point in continuing to support wx2.8, 3.0 is available in 'buntu repos for 14.04 LTS which is probably as far back as we need to support.
